File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    same_count = 0
    diff_count = 0
    target = ""
    
    for char in s:
        if target == "":
            target = char
        
        if target == char:
            same_count += 1
        else:
            diff_count += 1
        
        if same_count == diff_count:
            answer += 1
            same_count = 0
            diff_count = 0
            target = ""
    
    if same_count > 0:
        answer += 1
    
    return answer

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    let answer = 0;
    let sameCount = 0;
    let diffCount = 0;
    let target = '';
    
    for (let char of s) {
        if (target === '') {
            target = char;
        }
        
        if (char === target) {
            sameCount++;
        } else {
            diffCount++;
        }
        
        if (sameCount === diffCount) {
            answer++;
            sameCount = 0;
            diffCount = 0;
            target = '';
        }
    }
    
    if (sameCount > 0) {
        answer++;
    }
    
    return answer;
}